fiftyone/core/odm/database.py

```diff
@@ -1484,6 +1484,98 @@ def delete_runs(name, dry_run=False):
     )
 
 
+def get_indexed_values(
+    dataset, field_or_index, pipeline=None, values_only=False
+):
+    """Returns the values of the field for all samples in the given collection that are covered by the index.
+    Raises an error if the field is not indexed.
+
+    Args:
+        dataset: the dataset to query
+        field_or_index: the name of the indexed field to query or the index key for a compound index
+        pipeline: an optional pipeline to apply to the collection. For performance,
+            this should not include any fields not covered by the index.
+        values_only (False): whether to remove field names from the resulting list.
+            If True, the field names are removed and only the values will be returned as a list for each sample.
+            If False, the field names are preserved and the values will be returned as a dict for each sample.
+    Returns:
+        a list of values for the specified field or index keys for each sample sorted in the same order as the index
+    """
+    if field_or_index == "id":
+        if values_only:
+            return [
+                str(doc["_id"])
+                for doc in _iter_indexed_values(
+                    dataset, field_or_index, pipeline
+                )
+            ]
+        return [
+            {"id": str(doc["_id"])}
+            for doc in _iter_indexed_values(dataset, field_or_index, pipeline)
+        ]
+    if values_only:
+        # If values_only is True, we need to extract the values from the dict
+        if "_1_" not in field_or_index:
+            # Single field index
+            return [
+                doc[field_or_index]
+                for doc in _iter_indexed_values(
+                    dataset, field_or_index, pipeline
+                )
+            ]
+        return [
+            list(doc.values())
+            for doc in _iter_indexed_values(dataset, field_or_index, pipeline)
+        ]
+    return _iter_indexed_values(dataset, field_or_index, pipeline).to_list()
+
+
+def _iter_indexed_values(dataset, field_or_index, pipeline=None):
+    collection = dataset._sample_collection
+    if (
+        (field_or_index != "_id")
+        and (
+            field_or_index
+            not in (default_indexes := dataset._get_default_indexes())
+        )
+        and (
+            field_or_index not in (indexes := dataset.get_index_information())
+        )
+    ):
+        raise ValueError(
+            "Field '%s' is not indexed. Please create an index first or call `values()` instead"
+            % field_or_index
+        )
+    if field_or_index == "id" or field_or_index == "_id":
+        proj = {"_id": 1}
+        hint = {"_id": 1}
+    elif field_or_index in default_indexes:
+        if "_1" not in field_or_index:
```

Comment:
won't work if a default index with descending order is added.

Reply:
default indexes are hard coded right now and do not use descending order. Plus the qp panel also doesn't allow selecting descending order

Reply:
i understand that, im saying IF one was added.